cmake/compilerFlags.cmake

```diff
@@ -20,17 +20,39 @@ if ( MINGW OR UNIX OR MSYS ) # MINGW, Linux, APPLE, CYGWIN
         endif()
     endif()
 
+
     if (COMPILER_IS_GCC OR COMPILER_IS_CLANG)
 
+        # This fails under Fedora - MinGW - Gcc 8.3
+        if (NOT MINGW)
+            if (COMPILER_IS_GCC AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 8.0)
+                add_compile_options(-fstack-clash-protection -fcf-protection)
+            endif()
+
+            if (COMPILER_IS_GCC OR (COMPILER_IS_CLANG AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 3.7 ))
+                # is not available for clang 3.4.2. it appears to be present in clang 3.7.
+                add_compile_options(-fstack-protector-strong)
+            endif()
+        endif()
+
+        add_compile_options(-Wp,-D_GLIBCXX_ASSERTIONS)
+
+        if (CMAKE_BUILD_TYPE STREQUAL Release AND NOT APPLE)
+            add_compile_options(-Wp,-D_FORTIFY_SOURCE=2) # Requires to compile with -O2
+        endif()
+
         if(BUILD_WITH_COVERAGE)
-            set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -g ")
-            set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -O0")
-            set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fprofile-arcs")
-            set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -ftest-coverage")
+            add_compile_options(--coverage)
+            # TODO: From CMake 3.13 we could use add_link_options instead these 2 lines
             set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} --coverage")
+            set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} --coverage")
         endif()
 
-        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -Wcast-align -Wpointer-arith -Wformat-security -Wmissing-format-attribute -Woverloaded-virtual -W")
+        add_compile_options(-Wall -Wcast-align -Wpointer-arith -Wformat-security -Wmissing-format-attribute -Woverloaded-virtual -W)
+
+        # This seems to be causing issues in the Fedora_MinGW GitLab job

```

Comment:
Was this left on purpose?

Reply:
Yes, I thought it would be better to leave a comment explaining why that is added. Otherwise somebody else could be tempted to remove that guard.

Reply:
Sure, makes sense!